pkg/apis/openebs.io/scp/v1alpha1/storage-cohort.go

```diff
@@ -90,52 +90,28 @@ type ComponentStatus struct {
 
 	// NodeCondition is an array of current observed cohort's individual nodes conditions.
 	// +optional
-	NodeCondition []CohortNodeCondition `json:"nodeCondition,omitempty"`
+	NodeCondition map[string]ComponentConditionList `json:"nodeCondition,omitempty"`
```

Comment:
Instead of defining a new variable ComponentConditionList, can we try
NodeCondition map[string][]ComponentCondition `json:"nodeCondition,omitempty"` ?

Reply:
That is not supported by the controller-tool used to generate CRD yamls. Only primitive type array is supported For eg: `map[string][]string`.